actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -46,8 +48,166 @@ class TagBuilder # :nodoc:
         include CaptureHelper
         include OutputSafetyHelper
 
-        HTML_VOID_ELEMENTS = %i(area base br col embed hr img input keygen link meta param source track wbr).to_set
-        SVG_SELF_CLOSING_ELEMENTS = %i(animate animateMotion animateTransform circle ellipse line path polygon polyline rect set stop use view).to_set
+        def self.define_element(name, code_generator:, method_name: name.to_s.underscore)
+          code_generator.define_cached_method(method_name, namespace: :tag_builder) do |batch|
+            batch.push(<<~RUBY) unless instance_methods.include?(method_name.to_sym)
+              def #{method_name}(content = nil, escape: true, **options, &block)
+                tag_string(#{name.inspect}, content, escape: escape, **options, &block)
+              end
+            RUBY
+          end
+        end
+
+        def self.define_void_element(name, code_generator:, method_name: name.to_s.underscore, self_closing: false)
+          code_generator.define_cached_method(method_name, namespace: :tag_builder) do |batch|
+            batch.push(<<~RUBY)
+              def #{method_name}(content = nil, escape: true, **options, &block)
+                if content.present? || block
```

Comment:
```suggestion
                if content || block
```

Sorry for only seeing that one now. We should check if something was passed, doesn't matter if it's a blank string, or an empty string. If something was passed, it was passed.

Reply:
Done.